hera_pspec/tests/test_pspecbeam.py

```diff
@@ -0,0 +1,95 @@
+import unittest, os
+import nose.tools as nt
+import numpy as np
+import pyuvdata as uv
+from hera_pspec import pspecbeam
+
+
+# Get absolute path to data directory
+DATADIR = os.path.dirname( os.path.realpath(__file__) ) + "/../data/"
+
+
+class Test_DataSet(unittest.TestCase):
+
+    def setUp(self):
+        beamfile = DATADIR + 'NF_HERA_Beams.beamfits'
+        self.bm = pspecbeam.PSpecBeamUV(beamfile)
+        self.gauss = pspecbeam.PSpecBeamGauss(0.8,np.linspace(120.*10**6,128.*10**6,50))
+        pass
+
+    def tearDown(self):
+        pass
+
+    def runTest(self):
+        pass
+
+    def test_init(self):
+        beamfile = DATADIR + 'NF_HERA_Beams.beamfits'
+        bm = pspecbeam.PSpecBeamUV(beamfile)
+        pass
+
+    def test_UVbeam(self):
+        Om_p = self.bm.power_beam_int()
+        Om_pp = self.bm.power_beam_sq_int()
+        lower_freq = 120.*10**6
+        upper_freq = 128.*10**6
+        num_freqs = 50
+        scalar = self.bm.compute_pspec_scalar(lower_freq,upper_freq,num_freqs,stokes='I')
+
+        # Check array dimensionality
+        self.assertEqual(Om_p.ndim,1)
+        self.assertEqual(Om_pp.ndim,1)
+
+        # Check that errors are raised for other Stokes parameters
+        for stokes in ['Q', 'U', 'V', 'Z']:
+            nt.assert_raises(NotImplementedError, self.bm.power_beam_int, stokes=stokes)
+            nt.assert_raises(NotImplementedError, self.bm.power_beam_sq_int, stokes=stokes)
+            nt.assert_raises(NotImplementedError, self.bm.compute_pspec_scalar,lower_freq,upper_freq,num_freqs,stokes=stokes)
+
+        self.assertAlmostEqual(Om_p[0],0.078694909518866998)
```

Comment:
Where are these numbers from?

Reply:
Computed "by hand" in an iPython notebook, which is now also in the repo.